ui/src/router/components/form/components/router_config/route_card/RouteCard.js

```diff
@@ -13,16 +13,23 @@ import { SelectEndpointComboBox } from "../../../../../../components/form/endpoi
 import { EuiFieldDuration } from "../../../../../../components/form/field_duration/EuiFieldDuration";
 import { get } from "../../../../../../components/form/utils";
 import "./RouteCard.scss";
+import { EuiComboBoxSelect } from "../../../../../../components/form/combo_box/EuiComboBoxSelect";
 
 export const RouteCard = ({
   route,
+  protocol,
   endpointOptions,
   onChange,
   onDelete,
   errors,
 }) => {
   const [endpointsMap, setEndpointsMap] = useState({});
 
+  const upiServiceMethodOptions = [{
+    label: "/caraml.upi.v1.UniversalPredictionService/PredictValues",
```

Comment:
I did this with thought that it would support custom grpc too, but I probably shouldn't had done it for consistency.

Checked with aria, for UPI we will fix the service method and remove it totally from the card since its not configurable.

Reply:
removed service_method section and internally modify the obj